src/main/kotlin/com/exactpro/th2/check1/CollectorService.kt

```diff
@@ -84,7 +84,7 @@ class CollectorService(
 
         checkpointSubscriber = streamObservable.subscribeWith(CheckpointSubscriber())
 
-        ruleFactory = RuleFactory(configuration, streamObservable, eventBatchRouter)
+        ruleFactory = RuleFactory(configuration, streamObservable, eventBatchRouter, eventIdToLastCheckTask.keys)
```

Comment:
There might be a case when we create a rule using factory the chain exists but when we try to add the rule into the chain it is already cleaned and removed.

So, we must get the information about the chain existing only when we try to add the rule to a chain.
Maybe we should call the factory method inside `the compute` and pass the information about the chain existing as a parameter to the method for rule creation